src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -590,47 +688,143 @@ public override string ToString()
+
+            static Cci.IMethodDefinition synthesizeStaticConstructor(
+                ITokenDeferral module,
+                Cci.INamespaceTypeDefinition containingType,
+                MappedField dataField,
+                DataSectionStringField stringField,
+                Cci.IMethodDefinition bytesToStringHelper,
+                DiagnosticBag diagnostics)
+            {
+                var ilBuilder = new ILBuilder(
+                    module,
+                    new LocalSlotManager(slotAllocator: null),
+                    OptimizationLevel.Release,
+                    areLocalsZeroed: false);
+
+                // Push the `byte*` field's address.
+                ilBuilder.EmitOpCode(ILOpCode.Ldsflda);
+                ilBuilder.EmitToken(dataField, null, diagnostics);
+
+                // Push the byte size.
+                ilBuilder.EmitIntConstant(dataField.MappedData.Length);
+
+                // Call `<PrivateImplementationDetails>.BytesToString(byte*, int)`.
+                ilBuilder.EmitOpCode(ILOpCode.Call, -1);
+                ilBuilder.EmitToken(bytesToStringHelper, null, diagnostics);
+
+                // Store into the corresponding `string` field.
+                ilBuilder.EmitOpCode(ILOpCode.Stsfld);
+                ilBuilder.EmitToken(stringField, null, diagnostics);
 
-        public override Cci.INestedTypeDefinition AsNestedTypeDefinition(EmitContext context) => this;
+                ilBuilder.EmitRet(isVoid: true);
+                ilBuilder.Realize();
 
-        public override Cci.INestedTypeReference AsNestedTypeReference => this;
+                return new Cci.StaticConstructor(containingType, ilBuilder.MaxStack, ilBuilder.RealizedIL);
+            }
+        }
+
+        public Cci.IFieldDefinition Field => _fields[0];
+        public override string Name => _name;
+        public override Cci.ITypeDefinition ContainingTypeDefinition => _containingType;
+        public override Cci.TypeMemberVisibility Visibility => Cci.TypeMemberVisibility.Assembly;
+        public override Cci.ITypeReference GetBaseClass(EmitContext context) => _containingType.SystemObject;
+        public override IEnumerable<Cci.IFieldDefinition> GetFields(EmitContext context) => _fields;
+        public override IEnumerable<Cci.IMethodDefinition> GetMethods(EmitContext context) => _methods;
+        public override bool IsBeforeFieldInit => true;
+
+        private sealed class DataSectionStringField(
+            string name, Cci.INamedTypeDefinition containingType)
+            : SynthesizedStaticFieldBase(name, containingType)
+        {
+            public override ImmutableArray<byte> MappedData => default;
+            public override bool IsReadOnly => true;
+
+            public override Cci.ITypeReference GetType(EmitContext context)
+            {
+                return context.Module.GetPlatformType(Cci.PlatformType.SystemString, context);
+            }
+
+            public override string ToString()
```

Comment:
>ToString [](http://example.com/codeflow?start=35&length=8)

Where is `ToString()` used? Consider removing if not needed.

Reply:
It's part of the decompiled IL in tests apparently.